src/components/controller.cpp

```diff
@@ -161,26 +150,24 @@ bool controller::run(bool writeback, string snapshot_dst) {
   m_sig.attach(this);
 
   size_t started_modules{0};
-  for (const auto& block : m_modules) {
-    for (const auto& module : block.second) {
-      auto inp_handler = dynamic_cast<input_handler*>(&*module);
-      auto evt_handler = dynamic_cast<event_handler_interface*>(&*module);
+  for (const auto& module : m_modules) {
+    auto inp_handler = dynamic_cast<input_handler*>(&*module);
```

Comment:
Maybe, `std::dynamic_pointer_cast` can be used here (https://en.cppreference.com/w/cpp/memory/shared_ptr/pointer_cast). If so `m_inputhandlers` must store `std::shared_ptr`.

Reply:
This doesn't really have anything to do with this PR. We'll do this once we rework the event system which will require us to change how we handle input handlers anyway.

Reply:
Yeah you are right sorry.